components/compositing/touch.rs

```diff
@@ -22,10 +24,106 @@ const FLING_MIN_SCREEN_PX: f32 = 3.0;
 const FLING_MAX_SCREEN_PX: f32 = 4000.0;
 
 pub struct TouchHandler {
-    pub state: TouchState,
-    pub active_touch_points: Vec<TouchPoint>,
-    pub prevent_move: bool,
-    pub prevent_click: bool,
+    // The script thread is processing the flag for the touchmove operation.
+    pub handling_touch_move: bool,
+    pub current_sequence_id: u32,
+    touch_sequence_map: HashMap<u32, TouchSequenceInfo>,
+}
+
+struct TouchSequenceInfo {
+    // touch sequence state
+    state: TouchSequenceState,
+    // touch sequence active touch points
+    active_touch_points: Vec<TouchPoint>,
+    // Mark the end of the touch sequence
+    sequence_end: bool,
+    // Cancel click when touch move occurs or multiple touches on current thread.
+    cancel_click: bool,
+    // prevent default action in the touch sequence
+    prevent_default: bool,
+    // Once the first move has been processed by script, we can transition to
+    // non-cancellable events, and directly perform the pan without waiting for script.
+    first_move_processed: bool,
+    // Move operation waiting to be processed in the touch sequence
+    pending_touch_move_action: Option<TouchAction>,
```

Comment:
If we have `TouchSequenceState::WaitingForMoveHandler`, we could move this state into the enum variant.

Reply:
Let's leave the touch move for later, we need to think more about how to handle merging pending move actions.